src/garage/tf/algos/npo.py

```diff
@@ -156,13 +170,113 @@ def init_opt(self):
+
+        Args:
+            itr (int): Iteration number.
+            paths (list[dict]): A list of collected paths.
+
+        Returns:
+            numpy.float64: Average return.
+
+        """
+        # -- Stage: Calculate baseline
+        if self._flatten_input:
+            paths = [
+                dict(
+                    observations=(self._env_spec.observation_space.flatten_n(
+                        path['observations'])),
+                    actions=(
+                        self._env_spec.action_space.flatten_n(  # noqa: E126
+                            path['actions'])),
+                    rewards=path['rewards'],
+                    env_infos=path['env_infos'],
+                    agent_infos=path['agent_infos'],
+                    dones=path['dones']) for path in paths
+            ]
+        else:
+            paths = [
+                dict(
+                    observations=path['observations'],
+                    actions=(
+                        self._env_spec.action_space.flatten_n(  # noqa: E126
+                            path['actions'])),
+                    rewards=path['rewards'],
+                    env_infos=path['env_infos'],
+                    agent_infos=path['agent_infos'],
+                    dones=path['dones']) for path in paths
+            ]
+
+        if hasattr(self._baseline, 'predict_n'):
+            baseline_predictions = self._baseline.predict_n(paths)
+        else:
+            baseline_predictions = [
+                self._baseline.predict(path) for path in paths
+            ]
+
+        # -- Stage: Pre-process samples based on collected paths
+        samples_data = _functions.paths_to_tensors(paths, self.max_path_length,
+                                                   baseline_predictions,
+                                                   self._discount,
+                                                   self._gae_lambda)
+
+        # -- Stage: Run and calculate performance of the algorithm
+        undiscounted_returns = log_performance(
+            itr,
+            TrajectoryBatch.from_trajectory_list(self._env_spec, paths),
+            discount=self._discount)
+        self._episode_reward_mean.extend(undiscounted_returns)
+        tabular.record('Extras/EpisodeRewardMean',
+                       np.mean(self._episode_reward_mean))
+
+        samples_data['average_return'] = np.mean(undiscounted_returns)
+
+        self.log_diagnostics(samples_data)
+        logger.log('Optimizing policy...')
+        self.optimize_policy(itr, samples_data)
+        return samples_data['average_return']
+
+    def log_diagnostics(self, paths):
+        """Log diagnostic information.
+
+        Args:
+            paths (list[dict]): A list of collected paths.
+
+        """
+        logger.log('Logging diagnostics...')
+        self.policy.log_diagnostics(paths)
+        self._baseline.log_diagnostics(paths)
+
+    # pylint: disable=unused-argument
```

Comment:
Parameter `itr` is not used in the original `BatchPolopt` implementation, yet files in the tf/ directory still follows this common signature.  See this [workaround attempt](https://github.com/rlworkgroup/garage/blob/master/src/garage/np/algos/off_policy_rl_algorithm.py#L138). 

We can create a separate PR to refactor this.